File: app/src/test/java/de/gaffga/android/zazentimer/audio/VolumeDimmingTest.java
package at.priv.graf.zazentimer.audio;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class VolumeDimmingTest {

    @Test
    public void dimmingZero_mapsToFullVolume() {
        int dimming = 0;
        int volume = 100 - dimming;
        assertEquals(100, volume);
    }

    @Test
    public void dimmingFull_mapsToZeroVolume() {
        int dimming = 100;
        int volume = 100 - dimming;
        assertEquals(0, volume);
    }

    @Test
    public void dimmingFifty_mapsToFiftyVolume() {
        int dimming = 50;
        int volume = 100 - dimming;
        assertEquals(50, volume);
    }

    @Test
    public void volumeToMediaPlayerFloat_fullVolume() {
        int volume = 100;
        float f = volume / 100.0f;
        assertEquals(1.0f, f, 0.001f);
    }

    @Test
    public void volumeToMediaPlayerFloat_zeroVolume() {
        int volume = 0;
        float f = volume / 100.0f;
        assertEquals(0.0f, f, 0.001f);
    }

    @Test
    public void volumeToMediaPlayerFloat_halfVolume() {
        int volume = 50;
        float f = volume / 100.0f;
        assertEquals(0.5f, f, 0.001f);
    }

    @Test
    public void roundTrip_dimmingToVolumeToDimming() {
        for (int dimming = 0; dimming <= 100; dimming++) {
            int volume = 100 - dimming;
            int backToDimming = 100 - volume;
            assertEquals(dimming, backToDimming);
        }
    }
}

File: app/src/test/kotlin/at/priv/graf/zazentimer/audio/VolumeDimmingTest.kt
package at.priv.graf.zazentimer.audio

import org.junit.Assert.assertEquals
import org.junit.Test

class VolumeDimmingTest {
    @Test
    fun dimmingZero_mapsToFullVolume() {
        val dimming = 0
        val volume = 100 - dimming
        assertEquals(100, volume)
    }

    @Test
    fun dimmingFull_mapsToZeroVolume() {
        val dimming = 100
        val volume = 100 - dimming
        assertEquals(0, volume)
    }

    @Test
    fun dimmingFifty_mapsToFiftyVolume() {
        val dimming = 50
        val volume = 100 - dimming
        assertEquals(50, volume)
    }

    @Test
    fun volumeToMediaPlayerFloat_fullVolume() {
        val volume = 100
        val f = volume / 100.0f
        assertEquals(1.0f, f, 0.001f)
    }

    @Test
    fun volumeToMediaPlayerFloat_zeroVolume() {
        val volume = 0
        val f = volume / 100.0f
        assertEquals(0.0f, f, 0.001f)
    }

    @Test
    fun volumeToMediaPlayerFloat_halfVolume() {
        val volume = 50
        val f = volume / 100.0f
        assertEquals(0.5f, f, 0.001f)
    }

    @Test
    fun roundTrip_dimmingToVolumeToDimming() {
        for (dimming in 0..100) {
            val volume = 100 - dimming
            val backToDimming = 100 - volume
            assertEquals(dimming, backToDimming)
        }
    }
}
